ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -463,7 +464,29 @@ public static Path getCopyDestination(ReplChangeManager.FileInfo fileInfo, Path
     String[] subDirs = fileInfo.getSubDir().split(Path.SEPARATOR);
     Path destination = destRoot;
     for (String subDir: subDirs) {
-      destination = new Path(destination, subDir);
+      // If the directory is created by compactor, then the directory will have the transaction id also.
+      // In case of replication, the same txn id can not be used at target, as the txn with same id might be a
+      // aborted or live txn at target.
+      // In case of bootstrap load, we copy only the committed data, so the directory with only write id
+      // can be created. The validity txn id can be removed from the directory name.
+      // TODO : Support for incremental load flow. This can be done once replication of compaction is decided.
```

Comment:
TODO is not resolved ...currently we are doing compaction at target cluster also. If we chose to replicate the compaction from source, then this logic will change.

Reply:
The real question is whether there's a bug here, if compaction happens on the source as well as on the target. It would be better to clarify that this login needs to be worked again when incremental replication will be worked again.